File: ruby/bob/bob.rb
class Bob
  def hey(message)
    return 'Fine. Be that way!' if silence?(message)
    return 'Woah, chill out!'   if yelling?(message)
    return 'Sure.'              if question?(message)
    'Whatever.'
  end

  private

  def silence?(message)
    message =~ /\A\s*\z/
  end

  def yelling?(message)
    message !~ /[a-z]/
  end

  def question?(message)
    message =~ /\?\z/
  end
end

File: python/bob/bob.py
class Bob(object):
    def hey(_, message):
        if message is None or message.strip() == '':
            return 'Fine. Be that way!'
        if message.isupper():
            return 'Woah, chill out!'
        if message.endswith('?'):
            return 'Sure.'
        return 'Whatever.'
